src/benchmark_register.cc

```diff
@@ -461,16 +463,78 @@ int Benchmark::ArgsCnt() const {
   return static_cast<int>(args_.front().size());
 }
 
+void Benchmark::Init() {}
+
+namespace {
+
 //=============================================================================//
 //                            FunctionBenchmark
 //=============================================================================//
 
-void FunctionBenchmark::Run(State& st) { func_(st); }
+// The class used to hold all Benchmarks created from static function.
+// (ie those created using the BENCHMARK(...) macros.
+class FunctionBenchmark : public Benchmark {
```

Comment:
i'd be quite happy if these could be move out to their own header/source files (in src/).

Reply:
OK, maybe I shall also move benchmark's member function definitions to benchmark.cc?